### PR TITLE
Dates representation on inbox screen

### LINKED ISSUE #326: inbox dates completely wrong
I have emails in my inbox that on Gmail webmail show `10:49` (received a few minutes ago) but on iOS app it shows "18 Dec" or some other nonsensical date.

It seems all of them are wrong.

Let's fix this, add tests and use the following format:

- [ ] same day: `18:34`
- [ ] same year: `Jan 24`
- [ ] earlier year: `Jan 24, 2020`

### PR DESCRIPTION
**Pull request by ekievsky**
This PR fixes dates representation on inbox screen

close #326

----------------------------------

**Tests** :
- Tests added for date formatting

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [ ] addresses the issue it closes (if any)
- [ ] code is readable and understandable
- [ ] is accompanied with tests, or tests are not needed
- [ ] is free of vulnerabilities
- [ ] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

**Review by tomholub — APPROVED** [review]
Thanks!

**Comment by tomholub** [comment]
@Kharchevskyi could you see what happened to the tests?

**Comment by ekievsky** [comment]
@tomholub I see the error, it's on my side, I'll check it shortly.

**Comment by ekievsky** [comment]
Fixed

File: FlowCryptTests/ExtensionTests.swift [hunk 1]
```diff
@@ -69,13 +69,15 @@ extension ExtensionTests {
...
+        let sameDayDate = Date()
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "HH:mm"
         // Jan 24
         let sameYearDate = Date(timeIntervalSince1970: 1611506052)
         // Jan 24, 2020
         let otherYearDate = Date(timeIntervalSince1970: 1579883652)
         
-        XCTAssertEqual(DateFormatter().formatDate(sameDayDate), "18:34")
+        XCTAssertTrue(dateFormatter.date(from: DateFormatter().formatDate(sameDayDate)) != nil)
```
Comment (tomholub) [inline FlowCryptTests/ExtensionTests.swift:80]: This is not a robust way to test it - should at least test for a particular pattern like `/^\d\d:\d\d$/`

**Review by tomholub — APPROVED** [review]
Thank you - this fixes the issue

**Pull request merged**